src/coreclr/jit/morph.cpp

```diff
@@ -3720,7 +3726,8 @@ GenTree* Compiler::fgMorphMultiregStructArg(CallArg* arg)
             }
             else
 #elif defined(TARGET_LOONGARCH64)
-            if (arg->AbiInfo.StructFloatFieldType[inx] != TYP_UNDEF)
+            if ((arg->AbiInfo.StructFloatFieldType[inx] != TYP_UNDEF) &&
+                ((arg->AbiInfo.ByteSize == 8) || !varTypeIsGC(getSlotType(inx))))
```

Comment:
I had tested these code on LA.
For struct {float a; float b;} which using float register(s) and the size is 8, the elemCount is two but gctype's index should not greater than size/pointer-size within the layout.

Reply:
Makes sense. I think it would look better if we fixed the `varTypeIsGC` check instead of adding the `ByteSize` one:
```diff
-             if ((arg->AbiInfo.StructFloatFieldType[inx] != TYP_UNDEF) &&
-                ((arg->AbiInfo.ByteSize == 8) || !varTypeIsGC(getSlotType(inx)))) 
+             if ((arg->AbiInfo.StructFloatFieldType[inx] != TYP_UNDEF) &&
+                 !varTypeIsGC(getSlotType(offset / TARGET_POINTER_SIZE)))
```

Reply:
ok, thanks